File: examen-web-avanzadas/mascota.ts
export default class Mascota{
    private _id: number
    private _nombre: string
    private _fechaNacimiento: Date
    private _estaVacunado: boolean


    constructor(id: number, nombre: string, fecha: Date, esMacho: boolean) {
        this._id = id;
        this._nombre = nombre;
        this._fechaNacimiento = fecha;
        this._estaVacunado = esMacho;
    }

    get id(): number {
        return this._id;
    }

    set id(value: number) {
        this._id = value;
    }

    get nombre(): string {
        return this._nombre;
    }

    set nombre(value: string) {
        this._nombre = value;
    }

    get fechaNacimiento(): Date {
        return this._fechaNacimiento;
    }

    set fechaNacimiento(value: Date) {
        this._fechaNacimiento = value;
    }

    get estaVacunado(): boolean {
        return this._estaVacunado;
    }

    set estaVacunado(value: boolean) {
        this._estaVacunado = value;
    }
}

File: examen2/mascotas/src/main/kotlin/com/adriancoloma/mascotas/models/Mascota.kt
package com.adriancoloma.mascotas.models
import com.fasterxml.jackson.annotation.JsonFormat
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Table
import java.time.LocalDate


@Table(name = "mascota")
data class Mascota (
    @Id
    var id: Int? = null,
    var nombre: String? = null,
    @JsonFormat(pattern = "yyyy-MM-dd")
    var fechaNacimiento: LocalDate? = null,
    var estaVacunado: Boolean? = null,

)